src/services/StructuralMetadataUtils.test.js

```diff
@@ -0,0 +1,593 @@
+          begin: '00:00:11.23',
+          end: '00:08:00.00'
+        },
+        after: {
+          type: 'span',
+          label: 'Segment 2.1',
+          id: '123a-456b-789c-8d',
+          begin: '00:09:03.24',
+          end: '00:15:00.00'
+        }
+      };
+      const expected = [
+        {
+          type: 'div',
+          label: 'First segment',
+          id: '123a-456b-789c-1d'
+        },
+        {
+          type: 'div',
+          label: 'Sub-Segment 2.1',
+          id: '123a-456b-789c-6d'
+        }
+      ];
+      const value = smu.getValidHeadings(wrapperSpans, testData);
+      expect(value).toHaveLength(expected.length);
+      expect(value).toContainEqual({
+        type: 'div',
+        label: 'Sub-Segment 2.1',
+        id: '123a-456b-789c-6d'
+      });
+    });
+    test('no wrapping timespans after', () => {
+      const wrapperSpans = {
+        before: {
+          type: 'span',
+          label: 'Segment 2.1',
+          id: '123a-456b-789c-8d',
+          begin: '00:09:03.24',
+          end: '00:15:00.00'
+        },
+        after: null
+      };
+      const expected = [
+        {
+          type: 'div',
+          label: 'Sub-Segment 2.1',
+          id: '123a-456b-789c-6d'
+        }
+      ];
+      const value = smu.getValidHeadings(wrapperSpans, testData);
+      expect(value).toHaveLength(expected.length);
+      expect(value).toContainEqual({
+        type: 'div',
+        label: 'Sub-Segment 2.1',
+        id: '123a-456b-789c-6d'
+      });
+    });
+  });
+
+  describe('finds valid headings when either timespans before or after is null', () => {
+    test('wrapping timespan before in null', () => {
+      const wrapperSpans = {
+        before: null,
+        after: {
+          type: 'span',
+          label: 'Act 1',
+          id: '123a-456b-789c-3d',
+          begin: '00:10:00.00',
+          end: '00:15:00.00'
+        }
+      };
+      const value = smu.getValidHeadingForEmptySpans(
+        wrapperSpans,
+        testEmptyHeaderBefore
+      );
+      expect(value).toEqual([
+        { type: 'div', label: 'Scene 1', id: '123a-456b-789c-1d' }
+      ]);
+    });
+    test('wrapping timespan after is null', () => {
+      const wrapperSpans = {
+        before: {
+          type: 'span',
+          label: 'Act 1',
+          id: '123a-456b-789c-2d',
+          begin: '00:00:00.00',
+          end: '00:09:00.00'
+        },
+        after: null
+      };
+      const value = smu.getValidHeadingForEmptySpans(
+        wrapperSpans,
+        testEmptyHeaderAfter
+      );
+      expect(value).toEqual([
+        { type: 'div', label: 'Scene 2', id: '123a-456b-789c-3d' }
+      ]);
+    });
+  });
+
+  describe('validates order of begin and end times', () => {
+    test('begin time < end time', () => {
+      const begin = '00:00:10.30';
+      const end = '00:10:00.30';
+      const vlaue = smu.validateBeforeEndTimeOrder(begin, end);
+      expect(vlaue).toBeTruthy();
```

Comment:
Small typo in these tests for `vlaue`... if you want to fix great and if not no big deal.